app/src/main/java/com/example/tasky/common/data/TaskyStateImpl.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.tasky.common.data
+
+import com.example.tasky.common.domain.TaskyState
+import kotlinx.coroutines.flow.first
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TaskyStateImpl @Inject constructor(
```

Comment:
Not the most intuitive name IMO 😄 What do you think about `SessionStateManager`?